### ISSUE TITLE
Question re: booting without unlocking cfg in bios

### ISSUE THREAD

**Issue by bennals** [ThrRip/OpenCore-EFI-Laptop-Lenovo-Yoga-C740-14IML#28]
Hi,

Thanks for your work in this repositry.  I'm wondering if you might be able to suggest why [this efi](https://github.com/AZYLIO/Lenovo-Yoga-C740-81TD-OpenCore-Hackintosh) allows me to boot the Sonoma installer **without** needing to unlock cfg in the bios.

You state that unlocking cfg in bios is necessary for our C740s because the usual opencore quirks to work around the issue don't work for us.  That's what I found as well with my own attempts at creating an EFI following the Dortania guide, and also with your EFI.

Can you see anything in that linked EFI that might be a clue as to how they've acheived it?

Thanks,

Bill.

**Comment by bennals** [comment]
my apologies, I confused your statement about setting the value of DVMT pre-allocated in the bios with unlocking cfg (been staring at this thing for too long!).

In any case, I'm still interested if you can see anything in the linked efi that would enable me to boot into Sonoma (both the installer and after installation).

**Comment by ThrRip** [comment]
Hi Bill,

Disabling the CFG Lock and setting the DVMT pre-allocated value are **two separate things**.

The best case is that both can be done by adjusting settings in the BIOS, which is usually the case on non-OEM PCs. However, the Yoga C740 has the worst case, where neither of these configuration entries is available in the BIOS.

OpenCore has provided some configuration entries to deal with this.

- `AppleCpuPmCfgLock` and `AppleXcpmCfgLock` to get around with the CFG Lock
- `DeviceProperties -> Add -> PciRoot(0x0)/Pci(0x2,0x0) -> framebuffer-*` to set the DVMT pre-allocated value on the fly

Meanwhile, the Yoga C740 has _the worst of the worst case_. Even these entries don't work well. They may work sometimes, but the instability has led us to the idea of patching the BIOS.

---

The files in this repository aren't ready for higher minor versions of Sonoma (14.4 and above, as other users reported, while 14.3.1 and below are booting fine). That's probably why you've found the EFI files you linked can help you boot into Sonoma. I'm still looking for free time to work on the support for later versions of Sonoma, as well as Sequoia. I plan to get it done by the end of this year, but that's not a confirmed ETA.

**Comment by bennals** [comment]
Thanks for replying.

I took the plunge and used RU.efi to edit the bios and disable CFG Lock and increase DVMT pre-allocated to 128MB (heard that was important for 4K out to an external monitor - yet to try).

It was actually pretty straighforward and seems to have worked correctly.  I can now boot your EFI, and I also replaced the airportitlwm kext to the latest one released for Sonoma 14.4+, so now wifi and bluetooh are also working perfectly I think.  This is all on Sonoma 14.7.
